wandb/apis/public.py

```diff
@@ -1995,8 +1996,16 @@ def type(self):
 
     @property
     def name(self):
-        """The name by which the artifact was fetched."""
-        return self.artifact_name
+        """Stable name you can use to fetch this artifact."""
+        # TODO: All this logic should move to the backend.
+        if ":" not in self.artifact_name:
+            # this is a digest lookup
+            return self.artifact_name
+        artifact_collection_name = self.artifact_name.split(':')[0]
+        for alias in self._attrs["aliases"]:
+            if alias["artifactCollectionName"] == artifact_collection_name and re.match(r"^v\d+$", alias["alias"]):
+                return '%s:%s' % (artifact_collection_name, alias["alias"])
```

Comment:
We don't need to address it in this PR (behavior was like this before anyway), but wanted to call out that this behavior is different from that of local artifacts, where the `name` returns just the collection name.